syntax = "proto3";

message CodeResponse{
  string code = 1;
}

message Settings{
  int64 id = 1;
  int32 time_work = 2;
  int32 size_raw = 3;
  int32 type_input = 4;
  int32 type_output = 6;
  int32 number_shipments = 5;
  repeated string name_file = 7;
}

message GetSettingsRequest{
  int64 id = 1;
}

message GetSettingsResponse{
  Settings settings = 1;
}

service SettingsApi{
  rpc SettingsPost(Settings) returns (CodeResponse){}
  rpc SettingsGet(GetSettingsRequest) returns (GetSettingsResponse){}
  rpc SettingsUpdate(Settings) returns (CodeResponse){}
  rpc SettingsDelete(GetSettingsRequest) returns (CodeResponse){}
}